src/foraging_gui/TransferToNWB.py

```diff
@@ -409,9 +409,11 @@ class obj:
                         auto_train_stage_overridden=_get_field(obj, 'TP_auto_train_stage_overridden', index=i, default=np.nan),
 
                         # lickspout position
-                        lickspout_position_x=_get_field(obj, 'B_NewscalePositions', index=i, default=[np.nan] * 3)[0],
-                        lickspout_position_y=_get_field(obj, 'B_NewscalePositions', index=i, default=[np.nan] * 3)[1],
-                        lickspout_position_z=_get_field(obj, 'B_NewscalePositions', index=i, default=[np.nan] * 3)[2],
+                        lickspout_position_x=getattr(obj, 'B_StagePositions')[i]['x'] if len(getattr(obj, 'B_StagePositions', [])) > i else np.nan, 
+                        lickspout_position_y=getattr(obj, 'B_StagePositions')[i].get('y', np.nan) if len(getattr(obj, 'B_StagePositions', [])) > i else np.nan,
+                        lickspout_position_y1=getattr(obj, 'B_StagePositions')[i].get('y1', np.nan) if len(getattr(obj, 'B_StagePositions', [])) > i else np.nan,
+                        lickspout_position_y2=getattr(obj, 'B_StagePositions')[i].get('y12', np.nan) if len(getattr(obj, 'B_StagePositions', [])) > i else np.nan,
+                        lickspout_position_z=getattr(obj, 'B_StagePositions')[i].get('z', np.nan) if len(getattr(obj, 'B_StagePositions', [])) > i else np.nan,
                         
```

Comment:
@hanhou alex mentioned you would be a good person to ask how these lines work? I'm slightly confused by the _get_field function. Since the B_StagePositions are no longer a list of list coordinates but rather a list of dictionaries, does the approach I wrote work similarly?

Reply:
I just took care of this for the NWB packaging, which has to load either format of data.

https://github.com/AllenNeuralDynamics/NWB-Packaging-FiberPhotometry-Base-Capsule/pull/54/files

Reply:
I don't think we want to save both "y" and "y1" and "y2".

I'm not sure I follow what this code is trying to do.